File: Khizer/Day 3/euler.go
package main

import (
	"fmt"
)
func f(x float64 , y float64)float64{
	return x + y
}

func euler()[]float64{
	var x float64
	var y float64
	var b float64
	
    fmt.Println("Enter Value of x: ")
	fmt.Scanln(&x)
    fmt.Println("Enter value of y: ")
    fmt.Scanln(&y)
    fmt.Println("Enter step size (b): ")
    fmt.Scanln(&b)

	n := 5.0

	h := (b-x)/n
	arrX := []float64{x}
	arrY := []float64{y}
	// fmt.Println(arrX , arrY)
	fmt.Println(arrX[len(arrX) - 1] , b)
	// arrX = append(arrX, 10)
	for float64(arrX[len(arrX) - 1])< b{
		// fmt.Println("arr")

		arrX = append(arrX, float64(arrX[len(arrX) - 1]) + float64(h))
	}
	
	for index := range arrX{
		arrY = append(arrY , arrY[len(arrY) - 1] + (h * f(arrX[index] , arrY[index])))
	}
	return arrY
}

func main(){
	fmt.Println(euler())
}

File: Khizer/Day2/euler.py
# EULERS METHOD
#  Yn+1 = Yn + h * f(x,y)
# function f(x , y) = x + y

def f(x , y):
    return x + y 

def euler():
    x = float(input("Enter value of x (x < y)"))
    y = float(input("Enter value of y (y > x) "))
    b = float(input("Enter value of b (step size)"))
    n = 5
    h = (b - x)/n

    arrX = [x]
    arrY = [y]

    while arrX[len(arrX) - 1] < b:
        arrX.append(arrX[len(arrX) - 1] + h)

    for i in range(len(arrX) - 1):
        arrY.append(arrY[len(arrY) - 1] + (h * f(arrX[i] , arrY[i])))
    print(arrY)

euler()    